GenericsHomework/GenericsHomework/Node.cs

```diff
@@ -0,0 +1,64 @@
+﻿namespace GenericsHomework
+{
+    public class Node<T>
+    {
+        public Node(T value)
+        {
+            Value = value;
```

Comment:
```suggestion
            Value = value;
            Next = this;
```
It is a good idea to explicitly set next to `this` here in the constructor.

Reply:
Interesting, good to know thank you.